src/org/zaproxy/zap/extension/websocket/manualsend/WebSocketPanelSender.java

```diff
@@ -80,14 +94,103 @@ public void cleanup() {
     }
     
     private WebSocketProxy getDelegate(Integer channelId) {
-    	if (!connectedProxies.containsKey(channelId)) {
+    	if (closedProxies.containsKey(channelId)) {
     		logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
-    		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel") 
-    				+ " " + Constant.messages.getString("websocket.manual_send.fail"));
+    		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel")
+    				+ " " + Constant.messages.getString("websocket.manual_send.use_reopen"));
     	}
         return connectedProxies.get(channelId);
     }
 
+    /**
+     * Re-establish the connection
+	 *
+     * @param channelId connection to re-open
+     * @param newSecWebSocketKey New Sec-WebSocket-Key, null to use to generate new one
+	 * @param followRedirects true to follow redirections
+     * @param useSessionState useTrackingSessionState false for stateless http request
+	 * @throws IOException if connection refused
+	 * @throws RequestOutOfScopeException if url it's out of score ore ZAP runs to safe/protected mode
+	 * @throws IllegalArgumentException if an error occurred while trying to retrieve the HTTP handshake from history
+	 * @return the new channel id
+     */
+    public int reOpenChannel(Integer channelId, String newSecWebSocketKey, boolean followRedirects, boolean useSessionState) throws IllegalArgumentException, RequestOutOfScopeException, IOException {

```

Comment:
The exception `IllegalArgumentException` could be removed from the throws clause (it's unchecked).